### PR TITLE
parallel run fixes and lint changes

### PR DESCRIPTION
**Pull request by ebozduman**
Fix for sporadic functional tests/ build test sporadic failures introduced after parallel run change.
Some a bit of lint changes are also included.

### CODE REVIEW

File: Minio.Functional.Tests/Program.cs [hunk 1]
```diff
@@ -33,16 +33,17 @@ public static async Task Main(string[] args)
...
+        AppContext.SetSwitch("System.Net.Http.UseSocketsHttpHandler", false);
         var useAWS = Environment.GetEnvironmentVariable("AWS_ENDPOINT") is not null;
         if (Environment.GetEnvironmentVariable("SERVER_ENDPOINT") is not null)
         {
             endPoint = Environment.GetEnvironmentVariable("SERVER_ENDPOINT");
             var posColon = endPoint.LastIndexOf(':');
             if (posColon != -1)
             {
-                port = int.Parse(endPoint.Substring(posColon + 1, endPoint.Length - posColon - 1), NumberStyles.Integer,
+                port = int.Parse(endPoint.AsSpan(posColon + 1, endPoint.Length - posColon - 1), NumberStyles.Integer,
```
Comment (martijn00) [inline Minio.Functional.Tests/Program.cs:44]: This is not available in netstandard2.0

File: Minio.Functional.Tests/Program.cs [hunk 2]
```diff
@@ -33,16 +33,17 @@ public static async Task Main(string[] args)
...
                     CultureInfo.InvariantCulture);
-                endPoint = endPoint.Substring(0, posColon);
+                endPoint = endPoint[..posColon];
```
Comment (martijn00) [inline Minio.Functional.Tests/Program.cs:46]: This is not available in netstandard2.0

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 3]
```diff
@@ -2763,7 +2718,7 @@ internal static async Task ListenBucketNotificationsAsync_Test1(MinioClient mini
                         // data. This is required to match and convert json data
                         // "receivedJson" into class "ErrorResponse"
                         var len = "{'Error':".Length;
-                        var trimmedFront = receivedJson.Substring(len);
+                        var trimmedFront = receivedJson[len..];
```
Comment (martijn00) [inline Minio.Functional.Tests/FunctionalTest.cs:2721]: This is not available in netstandard2.0

**Review by martijn00 — CHANGES_REQUESTED** [review]
Looks good to me, but the new range syntax is not available in netstandard

**Comment by martijn00** [comment]
@ebozduman would be good to get this merged asap so i can rebase my PR and the other 2 PR's can too

**Comment by ebozduman** [comment]
@martijn00

I agree, but there are still some failures during build/functional tests. I think we need the flag to turn on/off parallel lrun flag, your suggestion, to be able to let those PRs pending to be merged just because the build tests fail when run in parallel, unrelated to the fixes/changes those PRs bring in.

**Comment by ebozduman** [comment]
I reviewed the changes one more time and I think I need to revisit this one, rebase and resolve the conflicts and definitely merge at least 2, likely medium priority changes in, if not all.

**Comment by ebozduman** [comment]
There is no need for this PR anymore.
Closing...

**Pull request closed without merging**